pkg/cloud/retry_manager.go

```diff
@@ -0,0 +1,50 @@
+package cloud
+
+import (
+	"github.com/aws/aws-sdk-go-v2/aws/retry"
+)
+
+const (
+	// retryMaxAttempt sets max number of EC2 API call attempts.
+	// Set high enough to ensure default sidecar timeout will cancel context long before we stop retrying.
+	retryMaxAttempt = 50
+)
+
+// newAdaptiveRetryer restricts attempts of API calls that recently hit throttle errors.
+func newAdaptiveRetryer() *retry.AdaptiveMode {
+	return retry.NewAdaptiveMode(func(ao *retry.AdaptiveModeOptions) {
+		ao.StandardOptions = append(ao.StandardOptions, func(so *retry.StandardOptions) {
+			so.MaxAttempts = retryMaxAttempt
+		})
+	})
+}
+
+// retryManager dictates the retry strategies of EC2 API calls.
+// Each mutating EC2 API has its own retryer because the AWS SDK throttles on a retryer object level, not by API name.
+// While default AWS accounts share request tokens between mutating APIs, users can raise limits for individual APIs.
+// Separate retryers ensures that throttling one API doesn't unintentionally throttle others with separate token buckets.
+type retryManager struct {
+	createVolumeRetryer                            *retry.AdaptiveMode
+	deleteVolumeRetryer                            *retry.AdaptiveMode
```

Comment:
np: we should probably use `aws.Retryer`, instead of `*retry.AdaptiveMode` directly.